src/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -493,9 +493,16 @@ public PsiElement bindToElement(@NotNull PsiElement element) throws IncorrectOpe
     else if (element instanceof PsiPackage) {
       bindToPackage((PsiPackage)element);
     }
+    else if (element instanceof PsiClass) {
+      bindToClass((PsiClass)element);
+    }
     return this;
   }
 
+  private void bindToClass(PsiClass element) {
+    handleElementRename(element.getName());
```

Comment:
`handleElementRename` method replaces Identifier with new one

so basically if class has full qualified name like `com.foo.Test`,
`Test` will be passed to rename function, `handleElementRename` made in such way that it will rename last Identifier only, so

if reference looks like `com.foo.Test`, `handleElementRename` would simply replace last, third Identifier `Test` with new Identifier with new name.

So it should work fine I hope. 
Java implementation may perform more checks https://github.com/JetBrains/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/PsiJavaCodeReferenceElementImpl.java#L545